### PR TITLE
fix: sort is not always an array

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix

### <samp>🤖 Generated by Copilot at 9b92bd8</samp>

Fixed a bug in `useChainRedirect.ts` that caused an invalid sort parameter to be passed to the API when redirecting from a remark prefix. Added a type check and a conversion logic for the `restOfQuery.sort` value.

### <samp>🤖 Generated by Copilot at 9b92bd8</samp>

> _`restOfQuery.sort`_
> _Array or string? Check first_
> _Autumn leaves no bugs_

### CODE REVIEW

File: composables/useChainRedirect.ts [hunk 1]
```diff
@@ -67,9 +67,13 @@ function getRedirectPathForPrefix({
...
     if (!isRemark.value && restOfQuery.sort) {
-      restOfQuery.sort = restOfQuery.sort.filter(
-        (value) => !value.startsWith('instance_'),
-      )
+      if (Array.isArray(restOfQuery.sort)) {
+        restOfQuery.sort = restOfQuery.sort.filter(
+          (value) => !value.startsWith('instance_'),
+        )
+      } else if (restOfQuery.sort.startsWith('instance_')) {
+        delete restOfQuery.sort
```
Comment (roiLeo) [inline composables/useChainRedirect.ts:75]: so are we using delete or not?

**Review by roiLeo — APPROVED** [review]
this part is still not easy to read, maybe we should refactor it or use better naming

**Comment by daiagi** [comment]
>this part is still not easy to read, maybe we should consider refactoring or use better naming

you're right.
please have a look again (although you approved already)

**Review by preschian — APPROVED** [review]

**Comment by prury** [comment]
i still get error tho, not sure if is the same error because it does not show any info, but is in the exact same line: 

image

**Comment by daiagi** [comment]
>i still get error tho, not sure if is the same error because it does not show any info, but is in the exact same line:


not the same error


seems to be metamask related:

metamask disabled:

image


enabled:

image

**Comment by prury** [comment]
> > i still get error tho, not sure if is the same error because it does not show any info, but is in the exact same line:
> 
> not the same error
> 
> seems to be metamask related:
> 
> metamask disabled:
> 
> image
> 
> enabled:
> 
> image



> > i still get error tho, not sure if is the same error because it does not show any info, but is in the exact same line:
> 
> not the same error
> 
> seems to be metamask related:
> 
> metamask disabled:
> 
> image
> 
> enabled:
> 
> image

ok, thank you and sorry for the trouble

**Pull request merged**